core/services/chainlink/config.go

```diff
@@ -170,6 +185,9 @@ func (c *Config) SetFrom(f *Config) (err error) {
 		err = multierr.Append(err, commonconfig.NamedMultiErrorList(err4, "Starknet"))
 	}
 
+	// the plugin should handle it's own defaults and merging
+	c.Aptos = f.Aptos
```

Comment:
This is unfortunately not so simple, because we also support passing multiple files that override one another. I can't think of a simple short-term fix, but this will have to be solved another way eventually regardless :thinking:

Reply:
could you concat the files together, then only do the parse once?

Reply:
Plain concat only works in special cases. We could write logic to crawl through the maps and set individual fields, but I'm not sure how easy that will be :thinking: